source/core/src/main/com/csse3200/game/components/player/PlayerStatsDisplay.java

```diff
@@ -1,75 +1,204 @@
+        hungerImage = new Image(ServiceLocator.getResourceService().getAsset("images/hunger_bar.png", Texture.class));
+
+        // Get the original width and height of the image
+        float barImageWidth = (float) (heartImage.getWidth() * 0.8);
+        float barImageHeight = (float) (heartImage.getHeight() * 0.5);
+
+        // Experience text maybe
+        CharSequence xpText = String.format("EXP: %d", 100);
+        xpLabel = new Label(xpText, skin, "large");
+
+        // Hunger text maybe
+        CharSequence hungerText = String.format("HGR: %d", 100);
+        hungerLabel = new Label(hungerText, skin, "large");
+
+        // Add all components to table
+        table.add(heartImage).size(barImageWidth, barImageHeight).pad(5);
+        table.add(healthLabel);
+        table.row();
+        table.add(xpImage).size(barImageWidth, (float) (barImageHeight * 1.25));
+        table.add(xpLabel);
+        table.row();
+        table.add(hungerImage).size(barImageWidth, barImageHeight * 2);
+        table.add(hungerLabel);
+        stage.addActor(table);
+    }
+
+    @Override
+    public void draw(SpriteBatch batch) {
+        // handled by stage
+    }
+
+    /**
+     * Updates the player's health on the ui.
+     *
+     * @param health player health
+     */
+    public void updatePlayerHealthUI(int health) {
+        CharSequence text = String.format("HP: %d", health);
+        System.out.println("Made it to this updateHealth function");
+        System.out.println(health);
+        healthLabel.setText(text);
+        int totalFrames = 11;
+
+        // Debugged and Developed with ChatGPT
+        // Calculate the frame index based on the current health
+        int maxHealth = entity.getComponent(CombatStatsComponent.class).getMaxHealth();
+        int frameIndex = totalFrames - 1 - (int) ((float) health / maxHealth * (totalFrames - 1));
+        frameIndex = Math.max(0, Math.min(frameIndex, totalFrames - 1));
+
+        // Set the current frame of the health bar animation
+        TextureRegion currentFrame = healthBarAnimation.getKeyFrame(frameIndex * 0.066f);
+        heartImage.setDrawable(new TextureRegionDrawable(currentFrame));  // Update the heartImage with the new frame
+    }
+    public void updatePlayerHungerUI(int hunger) {
+
+        CharSequence text = String.format("HGR: %d", hunger);
+        System.out.println("Made it to this updateHealth function");
+        System.out.println(hunger);
+        hungerLabel.setText(text);
+        int totalFrames = 11;
+
+        // Debugged and Developed with ChatGPT
+        // Calculate the frame index based on the current health
+        int maxHunger = entity.getComponent(CombatStatsComponent.class).getMaxHunger();
+        int frameIndex = totalFrames - 1 - (int) ((float) hunger / maxHunger * (totalFrames - 1));
+        frameIndex = Math.max(0, Math.min(frameIndex, totalFrames - 1));
```

Comment:
method call to both max and min unnecessary and hard to read; use Math.clamp instead

Reply:
will change this part in the second pull request